packages/kit-headless/src/components/pagination/pagination.tsx

```diff
@@ -1,231 +1,135 @@
-  showLastButton?: boolean;
-  activeClass?: string;
-  defaultClass?: string;
-  labels?: PaginationButtonLabels;
-}
-
-export interface PaginationButtonProps {
-  onClick$: PropFunction<() => void>;
   disabled?: boolean;
-  'aria-label': string;
-  'aria-current'?: boolean;
-  key?: string | number;
-  activeClass?: string;
-  defaultClass?: string;
-  value: PaginationButtonValue;
-  labels?: PaginationButtonLabels;
-}
+  customArrowTexts?: ArrowLabels;
 
-export type PaginationButtonValue =
-  | 'prev'
-  | 'next'
-  | 'first'
-  | 'last'
-  // | 'divider'
-  // | 'start-ellipsis'
-  // | 'end-ellipsis'
-  | number
-  | string;
-
-export interface PaginationButtonLabels {
-  first?: string;
-  last?: string;
-  next?: string;
-  prev?: string;
+  // styling
+  class?: string;
+  gap?: string;
+  defaultClass?: string;
+  selectedClass?: string;
+  dividerClass?: string;
 }
 
-const range = (start: number, end: number) => {
-  const length = end - start + 1;
-  return Array.from({ length }, (_, i) => start + i);
-};
-
-export const getPaginationButtons = (
-  page: number,
-  count: number,
-  labels: PaginationButtonLabels | undefined,
-  {
-    boundaryCount = 1,
-    siblingCount = 1,
-    hidePrevButton,
-    hideNextButton,
-    showFirstButton,
-    showLastButton,
-  }: PaginationOptions,
-): PaginationButtonValue[] => {
-  const startPages = range(1, Math.min(boundaryCount, count));
-  const endPages = range(Math.max(count - boundaryCount + 1, boundaryCount + 1), count);
-
-  const siblingsStart = Math.max(
-    Math.min(
-      page - siblingCount, // Natural start
-      count - boundaryCount - siblingCount * 2 - 1, // Lower boundary when page is high
-    ),
-    boundaryCount + 2, // Greater than startPages
+export const Pagination = component$<PaginationProps>((props) => {
+  const {
+    selectedPage,
+    totalPages,
+    onPageChange$,
+    // configuration
+    siblingCount,
+    boundaryCount,
+    hidePrevButton = false,
+    hideNextButton = false,
+    disabled = false,
+    customArrowTexts: {
+      previous: previousButtonLabel = 'PREV',
+      next: nextButtonLabel = 'NEXT',
+    } = {},
+    // styling
+    class: _class,
+    gap = '10px',
+    defaultClass,
+    selectedClass,
+    dividerClass,
+  } = props;
+
+  const isPrevButtonVisible = () => !hidePrevButton && selectedPage > 1;
+  const isNextButtonVisible = () => !hideNextButton && selectedPage !== totalPages;
+
+  const visibleItems = usePagination(
+    totalPages,
+    selectedPage,
+    siblingCount || 1,
+    boundaryCount || 1,
   );
 
-  const siblingsEnd = Math.min(
-    Math.max(
-      page + siblingCount, // Natural end
-      boundaryCount + siblingCount * 2 + 2, // Upper boundary when page is low
-    ),
-    endPages.length > 0 ? endPages[0] - 2 : count - 1, // Less than endPages
+  return (
+    <nav
+      role="navigation"
+      aria-label="pagination"
+      data-testid="pagination"
+      class={_class}
+      style={{
```

Comment:
```suggestion
      style={{
```

We want to add as little styles to the headless implementation as possible. Flex, align-items, and such should be added in the examples folder.

https://www.smashingmagazine.com/2022/05/you-dont-need-ui-framework/